data_adapter_oemof/build_datapackage.py

```diff
@@ -136,28 +151,84 @@ def get_foreign_keys(
                 else:
                     # Most likely the field may be a Timeseries in this case, but it is a scalar or unused.
                     pass
-        return new_foreign_keys
+        return new_foreignKeys
 
-    def save_datapackage_to_csv(self, destination):
+    def save_datapackage_to_csv(self, destination: str) -> None:
         """
         Saving the datapackage to a given destination in oemof.tabular readable format
-        :param destination:
-        :return:
+
+        Parameters
+        ----------
+        self: DataPackage
+            DataPackage to save
+        destination: str
+            String to where the datapackage save to. More convenient to use os.path. If last level of folder stucture
+            does not exist, it will be created (as well as /elements and /sequences)
+
+        Returns
+        -------
+        None if the Datapackage has been saved correctly (no checks implemented)
+
         """
+        # Check if filestructure is existent. Create folders if not:
+        elements_path = os.path.join(destination, "elements")
+        sequences_path = os.path.join(destination, "sequences")
+        if not os.path.exists(destination):
+            os.mkdir(destination)
+            os.mkdir(elements_path)
+            os.mkdir(sequences_path)
+        elif not os.path.exists(elements_path):
+            os.mkdir(elements_path)
+        elif not os.path.exists(sequences_path):
+            os.mkdir(sequences_path)
+
+        # Save elements to elements folder named by keys + .csv
+        for process_name, process_adapted_data in self.parametrized_elements.items():
+            process_adapted_data.to_csv(
+                os.path.join(elements_path, f"{process_name}.csv")
+            )
+
+        # Save Sequences to sequence folder named as keys + _sequence.csv
+        for process_name, process_adapted_data in self.parametrized_sequences.items():
+            process_adapted_data.to_csv(
+                os.path.join(sequences_path, f"{process_name}_sequence.csv")
+            )
+
+        # From saved elements and keys create a Package
+        package = Package(base_path=destination)
+        package.infer(pattern="**/*.csv")
+
+        # Add foreign keys from self to Package
+        for i, resource in enumerate(package.descriptor["resources"]):
+            if resource["name"] in self.foreignKeys.keys():
+                resource["schema"].update(
+                    {"foreignKeys": self.foreignKeys[resource["name"]]}
+                )
```

Comment:
Why don't you use `oemof.tabular.datapackage.building.infer_metadata` ?

Reply:
The implementation of infer metadata is only allowing for foreign keys with names "bus", "from_bus", "to_bus" and "profile" and might be sufficient but since we tried to keep it more open and implemented a sophisticated foreign keys approach this would reduce functionality a lot as Far as I understood.

Reply:
The `oemof.tabular.datapackage.building.infer_metadata` functionality also provides a different more general approach to derive foreign keys that I think is not strict enough for our Project, what do you think?